src/ray/stats/metric_exporter_client.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RAY_METRIC_EXPORTER_CLIENT_H
+#define RAY_METRIC_EXPORTER_CLIENT_H
+#include "ray/rpc/gcs_server/gcs_rpc_client.h"
+
+#include "ray/stats/metric.h"
+
+namespace ray {
+namespace stats {
+/// Interface class for abstract metrics exporter client.
+class MetricExporterClient {
+ public:
+  virtual void ReportMetrics(const MetricPoints &points) = 0;
+  virtual ~MetricExporterClient() = default;
+};
+
+/// Default stdout exporter client can log metrics info for debug.
+class StdoutExporterClient : public MetricExporterClient {
+ public:
+  void ReportMetrics(const MetricPoints &points) override;
+};
+
+/// The decoration mode is that the user can apply it by configuring different
+/// combinations.
+/// Usage:
+/// std::shared_ptr<MetricExporterClient> exporter(new StdoutExporterClient());
+/// std::shared_ptr<MetricExporterClient> gcs_exporter_client(
+///         new GcsExporterClient(exporter, gcs_rpc_client));
```

Comment:
Can you change the example to use dashboard exporter instead of GCS exporter? We will directly export metrics to dashboard agents, not GCS